src/System.Runtime.Numerics/src/System/Numerics/BigIntegerCalculator.PowMod.cs

```diff
@@ -0,0 +1,432 @@
+            }
+        }
+
+        public static uint Pow(uint value, uint power, uint modulus)
+        {
+            // The 32-bit modulus pow method for a 32-bit integer
+            // raised by a 32-bit integer...
+
+            uint result = PowCore(power, modulus, value, 1);
+
+            return result;
+        }
+
+        public static uint Pow(uint[] value, uint power, uint modulus)
+        {
+            Debug.Assert(value != null);
+
+            // The 32-bit modulus pow method for a big integer
+            // raised by a 32-bit integer...
+
+            uint v = Remainder(value, modulus);
+
+            uint result = PowCore(power, modulus, v, 1);
+
+            return result;
+        }
+
+        public static uint Pow(uint value, uint[] power, uint modulus)
+        {
+            Debug.Assert(power != null);
+
+            // The 32-bit modulus pow method for a 32-bit integer
+            // raised by a big integer...
+
+            uint result = PowCore(power, modulus, value, 1);
+
+            return result;
+        }
+
+        public static uint Pow(uint[] value, uint[] power, uint modulus)
+        {
+            Debug.Assert(value != null);
+            Debug.Assert(power != null);
+
+            // The 32-bit modulus pow method for a big integer
+            // raised by a big integer...
+
+            uint v = Remainder(value, modulus);
+
+            uint result = PowCore(power, modulus, v, 1);
+
+            return result;
+        }
+
+        private static uint PowCore(uint[] power, uint modulus,
+                                    ulong value, ulong result)
+        {
+            // The 32-bit modulus pow algorithm for all but
+            // the last power limb using square-and-multiply.
+
+            for (int i = 0; i < power.Length - 1; i++)
+            {
+                uint p = power[i];
+                for (int j = 0; j < 32; j++)
+                {
+                    if ((p & 1) == 1)
+                        result = (result * value) % modulus;
+                    value = (value * value) % modulus;
+                    p = p >> 1;
+                }
+            }
+
+            return PowCore(power[power.Length - 1], modulus, value, result);
+        }
+
+        private static uint PowCore(uint power, uint modulus,
+                                    ulong value, ulong result)
+        {
+            // The 32-bit modulus pow algorithm for the last or
+            // the only power limb using square-and-multiply.
+
+            while (power != 0)
+            {
+                if ((power & 1) == 1)
+                    result = (result * value) % modulus;
+                if (power != 1)
+                    value = (value * value) % modulus;
+                power = power >> 1;
+            }
+
+            return (uint)(result % modulus);
+        }
+
+#if DEBUG
+        private const int ReducerThreshold = 8;
+#else
+        private const int ReducerThreshold = 32;
+#endif
```

Comment:
My concern is that a lot of our testing is done against debug builds, and I get a bit concerned any time I see debug vs release settings changing behavior, which this essentially does.  What would the downside be to just always making it 32?  Are we unable to write tests that appropriately exercise the reducer and still complete in a reasonable amount of time?